astropy/io/fits/tests/test_hdulist.py

```diff
@@ -938,16 +1004,14 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         with unicode content that is not actually a FITS file. See:
         https://github.com/astropy/astropy/issues/5594#issuecomment-266583218
         """
-        import codecs
         filename = self.temp('not-fits-with-unicode.fits')
-        with codecs.open(filename, mode='w', encoding='utf=8') as f:
+        with open(filename, mode='w', encoding='utf=8') as f:
             f.write('Ce\xe7i ne marche pas')
 
         # This should raise an OSError because there is no end card.
-        with pytest.raises(OSError):
-            with pytest.warns(AstropyUserWarning, match=r'non-ASCII characters '
-                              r'are present in the FITS file header'):
-                fits.open(filename)
+        with pytest.raises(OSError, match='No SIMPLE card found, this file '
+                           'does not appear to be a valid FITS file'):
+            fits.open(filename)
```

Comment:
With
```
fits.open(filename, ignore_missing_simple=True)
```
this should still raise the `AstropyUserWarning` once it encounters the non-ASCII char, right? Could this case be tested additionally?

Reply:
I don't think so, since this is an invalid file the error must be raised instead of the warning.
I guess one question is to know is there is another test with non-ASCII char in a valid header/file, I can check that later.

Reply:
Yes, would be good. My point was, `ignore_missing_simple=True` is supposed to allow people to open otherwise valid files that are _only_ missing the SIMPLE card, but an otherwise broken should still raise an exception (which would rather not be the one about missing SIMPLE). But I don't know the details of #5594 and if it still requires a dedicated test.